### ISSUE TITLE
Jubilee 1 Project Vote

### ISSUE THREAD

**Issue by JordanRL** [PHPJubilee/Project#8]
## List to Choose From

(This list is in the order they were suggested.)
- (Score: 7) [Fermat](https://github.com/JordanRL/Fermat)
- (Score: -6) [MagpieRSS](https://github.com/kellan/magpierss)
- (Score: -3) [cPchart](https://github.com/szymach/c-pchart)
- (Score: 4) [PHP Collection](https://github.com/schmittjoh/php-collection)
- (Score: 1) [Kint](https://github.com/raveren/kint)
- (Score: 3) [gitlist](https://github.com/klaussilveira/gitlist)
## Voting

You may cast one :+1: or :-1: for each possible project. You can also choose not to vote one way or the other for any or all of the items on the list.

**Only your most recent vote is counted.** You can post with different votes as many times as you want, but only your most recent vote(s) count toward the score.
## Discussion

You can also provide any reasoning you wish to share as to why you're voting the way you are, or discuss back and forth with others your reasoning.
## Eligibility

Only members of PHPJubilee who are on at least one team may vote. Voting is **not** restricted to those who were members before voting started, so new members may vote as they join.
## Deadline

Voting closes October 30th, unless we reach clear consensus before then.

**Comment by SketchNI** [comment]
:thumbsup: for gitlist

**Comment by JordanRL** [comment]
:+1: Fermat. Being the maintainer, I obviously see some positives to this option. But I also think it could be helpful if our first sprint was for a project where the maintainer is part of the sprint.

:+1: PHP Collection. An interesting project on its own, this one would also expose everyone on the team to some more complex data structures.

:+1: gitlist. This project is almost the quintessential example of what this team is for.

:-1: MagpieRSS. Too big of a project IMO.

**Comment by martinph** [comment]
:+1: gitlist 

:+1: fermat

:-1: magpierss

As useful as magpierss has been for people over the years, I think it's time for the project to fade away. There's far better and more active rss writers and readers out there.

**Comment by SketchNI** [comment]
On second thoughts, as well as my :+1: for gitlist,
:+1: Fermet
:+1: PHP Collection

**Comment by bayne** [comment]
:+1: Fermat - Having the first sprint where @JordanRL is the maintainer would provide the most valuable feedback on how to manage the jubilee from the maintainer side. A retrospective discussion after this sprint might be helpful for guiding future sprints.

:-1: gitlist - I feel saving this for a future sprint might best

**Comment by dersam** [comment]
:+1: Fermat: As @bayne said, this will give us the most information on how to manage the jubilee. Also :+1: on doing a post-sprint retrospective.

:-1: gitlist- While it would be a good fit for the Jubilee, I don't think it's best to jump right in with it being the first one.
:-1: magpie: I don't think we should be looking to resurrect dead projects

**Comment by JordanRL** [comment]
Current totals have been updated. Remember, voting is still open until October 30th.

**Comment by saji89** [comment]
My list of vote, in decreasing order of preference is:
1. :+1: `Fermat` - As many others said above, its good to take up a project with the maintainer( @JordanRL ) among us, initially.
2. :+1: `kint` - Looks pretty good and useful.
3. :+1: `PHP Collection` - Interesting project, that will introduce the area not very well explored in the PHP world (generally), data structures. 
4. :+1: `gitlist` - Awesome project. We can probably take this up later, if possible. It might need more understanding about git files, so we might require more time on this.
5. :-1: `MagpieRSS` - It seems to be a dead project at the moment, maybe we could take it up once we're accustomed and decided upon the procedures of our team, and the skillset we have.
6. :-1: `cPchart` - Looks the least interesting at the moment.

**Comment by naroga** [comment]
:+1: Fermat
:+1: PHP Collection
:+1: gitlist
:-1: kint
:-1: MagpieRSS
:-1: cPchart

**Comment by lancepioch** [comment]
List of preference from greatest to least:
1. gitlist :+1: 
2. kint :+1: 
3. PHP Collection :+1: 
4. Fermat :-1: 
5. c-pChart :-1: 
6. MagpieRSS :-1:

**Comment by naroga** [comment]
Fermat is by far the most voted project, but it looks like it needs little work. It's very specific in what it does (it's not a huge library), it's already PSR1/2/4 compliant, already uses composer, there are unit tests (although they might need more coverage), and no open issues.

@JordanRL, can you suggest more work to be done in this library? Or should we maybe elect two libraries for the first sprint?

**Comment by JordanRL** [comment]
@naroga Fermat doesn't have unit tests, it just has a tests folder and bootstrap file. There is some refactoring that it could use, and if it wins then I'm going to reorganize a few things to make it easier, and open up tickets for us as well. I'm currently the only contributor, so I haven't been ticketing my work (since it's pre-release right now).

I was actually in the middle of adding a whole vector math section to the library, which would be both interesting and different.

I do get your point though. As it is right now it doesn't look like there's a sprint's worth of work there, however:
1. We have lots of people that are enthusiastic, which is awesome. But this is a freetime project. I fully expect that there will be people who don't contribute at all for this first sprint, or on any given sprint, just because of availability.
2. I haven't been documenting much of the bugs, todo's, work, etc. for Fermat because I was developing it in response to needs in Newton. However, I do have plenty of things which need to be done... I just haven't been making issues for them.
3. One of the things about working on a project where one of us are the maintainer is that this team will have more leeway for doing things like refactors, changes to implementation, plain improvements, etc. Some OSS projects don't appreciate that as much, and the focus of this team is to eliminate technical debt, but there are also things which we can improve in all the projects we work on.

Basically, each project will have areas that it is strong in, and areas where it needs work. Part of this vote should be looking at the projects, what those projects need, and deciding if that's the sort of work you want to do next sprint.

---

EDIT: Just a quick note. One of the interesting things we'll have to look at if we do Fermat is getting the stats extension working in the PHP7 build on Travis-CI. Right now Travis-CI won't build ext-stats on PHP 7.

**Comment by naroga** [comment]
@JordanRL Great points. I've voted for Fermat already, those were just a general wonderment.

**Comment by naroga** [comment]
As discussed on the IRC channel (#PHPJubilee at irc.freenode.net, everyone should join) today, I'll withdraw my votes for gitlist and phpcollection. This should give Fermat a more comfortable margin.

**Comment by saji89** [comment]
@JordanRL @naroga Good god. We now have an IRC channel too? I thought we're using Gitter. :confused:

**Comment by JordanRL** [comment]
@naroga Alright, I've updated your votes in the totals.

**Comment by NoodlesNZ** [comment]
:+1: Fermat

**Comment by JordanRL** [comment]
**Vote totals current as of this post.**
